torch/csrc/jit/passes/utils/memory_dag.h

```diff
@@ -70,13 +72,11 @@ class TORCH_API MemoryDAG {
     }
 
     // Record all memory locations from group `a`
-    std::unordered_set<const Element*> memoryLocations;
+    MemoryLocations memoryLocations;
```

Comment:
Seems like this never returns true because you're just building up a big memoryLocations set and not doing anything with it. Since intersection is fast now, we probably don't need the short-circuiting implementation and it's fine to build up a's memory locations, b's memory locations, and return `memoryLocationOverlap`

Reply:
... Whups.

Reply:
I think it's fine to keep the short-circuiting implementation. `Intersection` isn't really the bottleneck, the bottleneck is memory allocation.